pkg/process/net/common.go

```diff
@@ -162,6 +162,37 @@ func (r *RemoteSysProbeUtil) GetConnections(clientID string) (*model.Connections
 	return conns, nil
 }
 
+// GetPing returns the results of a ping to a host
+func (r *RemoteSysProbeUtil) GetPing(clientID string, host string, count int, interval time.Duration, timeout time.Duration) ([]byte, error) {
+	req, err := http.NewRequest("GET", fmt.Sprintf("%s/%s?client_id=%s&count=%d&interval=%d&timeout=%d", pingURL, host, clientID, count, interval, timeout), nil)
+	if err != nil {
+		return nil, err
+	}
+
+	req.Header.Set("Accept", "application/json")
+	resp, err := r.httpClient.Do(req)
+	if err != nil {
+		return nil, err
+	} else if resp.StatusCode == http.StatusBadRequest {
+		body, err := io.ReadAll(resp.Body)
+		defer resp.Body.Close()
```

Comment:
nit: is the `defer` necessary here? I mean can we just `resp.Body.Close()` without the `defer`? Same question for the other `defer resp.Body.Close()` statements.

Reply:
Good catch, I'm not really sure what I was doing here. This should be done right after we receive the response.